include/hip/hcc_detail/ocml_math_fwd.h

```diff
@@ -0,0 +1,431 @@
+__attribute__((const))
+double __ocml_rint_f64(double);
+__attribute__((const))
+double __ocml_rlen3_f64(double, double, double);
+__attribute__((const))
+double __ocml_rlen4_f64(double, double, double, double);
+__attribute__((const))
+double __ocml_round_f64(double);
+__attribute__((pure))
+double __ocml_rsqrt_f64(double);
+__attribute__((const))
+double __ocml_scalb_f64(double, double);
+__attribute__((const))
+double __ocml_scalbn_f64(double, int);
+__attribute__((const))
+int __ocml_signbit_f64(double);
+double __ocml_sincos_f64(double, __attribute__((address_space(5))) double*);
+double __ocml_sincospi_f64(double, __attribute__((address_space(5))) double*);
+double __ocml_sin_f64(double);
+__attribute__((pure))
+double __ocml_sinh_f64(double);
+double __ocml_sinpi_f64(double);
+__attribute__((const))
+double __ocml_sqrt_f64(double);
+double __ocml_tan_f64(double);
+__attribute__((pure))
+double __ocml_tanh_f64(double);
+double __ocml_tgamma_f64(double);
+__attribute__((const))
+double __ocml_trunc_f64(double);
+double __ocml_y0_f64(double);
+double __ocml_y1_f64(double);
+
+// BEGIN INTRINSICS
+__attribute__((const))
+double __ocml_add_rte_f64(double, double);
+__attribute__((const))
+double __ocml_add_rtn_f64(double, double);
+__attribute__((const))
+double __ocml_add_rtp_f64(double, double);
+__attribute__((const))
+double __ocml_add_rtz_f64(double, double);
+__attribute__((const))
+double __ocml_sub_rte_f64(double, double);
+__attribute__((const))
+double __ocml_sub_rtn_f64(double, double);
+__attribute__((const))
+double __ocml_sub_rtp_f64(double, double);
+__attribute__((const))
+double __ocml_sub_rtz_f64(double, double);
+__attribute__((const))
+double __ocml_mul_rte_f64(double, double);
+__attribute__((const))
+double __ocml_mul_rtn_f64(double, double);
+__attribute__((const))
+double __ocml_mul_rtp_f64(double, double);
+__attribute__((const))
+double __ocml_mul_rtz_f64(double, double);
+__attribute__((const))
+double __ocml_div_rte_f64(double, double);
+__attribute__((const))
+double __ocml_div_rtn_f64(double, double);
+__attribute__((const))
+double __ocml_div_rtp_f64(double, double);
+__attribute__((const))
+double __ocml_div_rtz_f64(double, double);
+__attribute__((const))
+double __ocml_sqrt_rte_f64(double);
+__attribute__((const))
+double __ocml_sqrt_rtn_f64(double);
+__attribute__((const))
+double __ocml_sqrt_rtp_f64(double);
+__attribute__((const))
+double __ocml_sqrt_rtz_f64(double);
+__attribute__((const))
+double __ocml_fma_rte_f64(double, double, double);
+__attribute__((const))
+double __ocml_fma_rtn_f64(double, double, double);
+__attribute__((const))
+double __ocml_fma_rtp_f64(double, double, double);
+__attribute__((const))
+double __ocml_fma_rtz_f64(double, double, double);
+
+__attribute__((const))
+double __llvm_amdgcn_rcp_f64(double) __asm("llvm.amdgcn.rcp.f64");
+__attribute__((const))
+double __llvm_amdgcn_rsq_f64(double) __asm("llvm.amdgcn.rsq.f64");
+// END INTRINSICS
+// END DOUBLE
+
```

Comment:
FWIW, OCML provides half precision variants of the math functions.  Should they be included here for later use?

Reply:
They are already used in the __half / __half2 implementations. If this gets merge moving them into this header may be worth considering. Not a concern for this PR.